### PR TITLE
feat(service-portal): new design

### PR DESCRIPTION
**Pull request by disaerna**
# New design for service portal 🎉 

## What

Implementing new design for Service Portal (Mínar síður). 
**Warning .... this is a **HUGE** PR**

Changes can be seen and tested here: [https://service-portalnew-design-demo-beta.dev01.devland.is/minarsidur](https://service-portalnew-design-demo-beta.dev01.devland.is/minarsidur)

- Sidebar fixed to left
- New design for active items
- Collapsable sidebar with icons
- Collapsable sidebar has submenu modal on hover
- Collapsable sidebar items
- New design for UserMenu
- Refactor how user sees actors and when viewing as an actor
- Add language changer next to userMenu like main web for desktop
- Layout changes on sidebar toggle + finance part has larger grid layout
- Lock icon appears on locked nav items and returns lock screen instead of disabling sidebar item
- Badge appears on "Pósthólf" when user has unread documents

- UI core library modified:
* Input, Select and Datepicker now has XS option, with label outside the box
* ActionCard - Added prop "headingVariant" and 'h3' and 'h4' as available values, defaults to 'h3' and displays as 'h3'
* UserAvatar - Added a medium size option and two props, 'color' that has two options, purple or blue and 'isUserMenu' that uses different css if true
* ArrowLink - Added width: max-content to the css to prevent breaking in two lines on some screens

## Why

New Design !

### Sidebar: 

Screenshot 2022-01-05 at 09 06 45Screenshot 2022-01-05 at 11 42 13

### UserMenu:

Mobile:
Screenshot 2022-01-05 at 09 08 48

Desktop:
Screenshot 2022-01-05 at 09 07 35

Desktop - delegation: 
Screenshot 2022-01-05 at 09 07 53

### Mobile Menu

Screenshot 2022-01-05 at 09 08 27

Screenshot 2022-01-05 at 09 08 40

### Layout:
 Screenshot 2022-01-05 at 09 07 20

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by sigruntg — APPROVED** [review]

File: apps/service-portal/src/components/Layout/constants.ts
Comment (disaerna) [reply apps/service-portal/src/components/Layout/constants.ts]: Woah 😬  - I'll fix

File: libs/service-portal/endorsements/src/screens/Endorsements/Endorsements.tsx
Comment (arnorhs) [inline libs/service-portal/endorsements/src/screens/Endorsements/Endorsements.tsx:1]: isn't this something that has been removed on main?

File: libs/service-portal/education-career/src/screens/EducationCareer/EducationCareer.tsx
Comment (arnorhs) [inline libs/service-portal/education-career/src/screens/EducationCareer/EducationCareer.tsx:21]: has the image been removed as well?

**Review by arnorhs — CHANGES_REQUESTED** [review]

File: libs/service-portal/endorsements/src/screens/Endorsements/Endorsements.tsx
Comment (disaerna) [reply libs/service-portal/endorsements/src/screens/Endorsements/Endorsements.tsx]: Yes correct! - Removed ✔️

File: libs/service-portal/education-career/src/screens/EducationCareer/EducationCareer.tsx
Comment (disaerna) [reply libs/service-portal/education-career/src/screens/EducationCareer/EducationCareer.tsx:21]: Yes all images on these landing screens are removed

**Review by hinriksteinar — APPROVED** [review]

File: libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx:239]: ?

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserMenu.tsx:42]: Is this needed?
`</nitpick>`

File: libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx:31]: These links don't work in other apps (application-system-form) since they use `react-router-dom` links relative to `/minarsidur`. May need to use normal `<a>` links with full path, at least in other apps.

Comment (eirikurn) [reply libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx:31]: Can test here: https://service-portalnew-design-demo-beta.dev01.devland.is/umsoknir/example

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserMenu.tsx:12]: This PR is missing a language picker in the application system form. Space is more tight there, which is why I liked having the language picker (consistently) in the user menu.

Thinking was that the user is unlikely to change this part of their user profile many times so it doesn't need to take space on every screen when you're logged in.

image

But maybe there is space for the to fall into one more line, vertically centered.

File: libs/shared/components/src/auth/UserMenu/UserDelegations.tsx
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserDelegations.tsx:82]: Design comment:

Not loving the design of the delegation list during an active delegation.

* The purple card background is so light I can barely see it, and I think it's weird that the Avatar circle disappears into it.
* The active delegation is still in the list, but shows no special state indicating that it is active and can't be clicked.

image

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (eirikurn) [inline libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:126]: I can't find these pages any more in the navigation:

image

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (eirikurn) [inline apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: Design comment:

I wish the collapsed state could be a bit tighter, there's just soo much horizontal whitespace around the small navigation icons. This is a big waste on tablet sized window sizes but sticks out to me personally on all screen sizes.

image

I imagine it's "beta"'s fault. Could we collapse the beta below the logo instead?

**Review by eirikurn — COMMENTED** [review]
Awesome job! 🎉🎉🎉 

Reviewed code owner files and gave some feedback on the design.

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: I'll take this to Keli, the designer

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:126]: Yes correct, it's now only available under UserMenu

File: libs/shared/components/src/auth/UserMenu/UserDelegations.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserDelegations.tsx:82]: Whops - The current user should not be in the list anymore and the avatar should be different color - thanks I'll fix it!
But the purple color is to show you your main user. I misunderstood it when we spoke the other day. The heading displaying the current user and main user below is suppose to be enough to show active delegation but i'll take this one to the designer as well.

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:12]: I see - The language part was taken from UserMenu and the landing was to make it more like the main web, spoke to @hrefnalind about this. I totally forgot about this part. I'll view this and put it back in under condition?

File: libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx:31]: Ah - I'll fix, thanks

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:42]: No 😅  I'll fix

File: libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx:239]: It's terrible - don't look (but I can screen record if you like). I'm waiting on "NumberFormat" package to release a fix so that I can change the size variant for numbered inputs. This beautiful hack is only temporary 😬 

You can see the problem here: https://github.com/s-yadav/react-number-format/pull/577

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:126]: Interesting 👍

File: libs/shared/components/src/auth/UserMenu/UserDelegations.tsx
Comment (eirikurn) [reply libs/shared/components/src/auth/UserMenu/UserDelegations.tsx:82]: Ok sure. Maybe the fixes you mention will be enough for it to make more sense to me.

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (eirikurn) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx:12]: Yeah, I think it is more important for it to be consistent with other places where a user is logged in than the main web. But if we can have it globally consistent, then great I guess? :)

File: libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx
Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx:239]: Haha, you're right. It's terrible! 😅

Do we want to get rid of all of the large inputs in the service portal? I thought it was only in the places where we wanted to save space or take less attention. I would argue that this is the primary thing on this screen so it can use the "standard" large input.

Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx:239]: Also, there's a layout regression when showing a validation error:
image

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (eirikurn) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: Also, can we have the consistent whitespace on the left and right side of the main content. Currently it uses `margin-left: 25%;` and `16%` which looks weird to me and wastes space on smaller screens. Maybe we can use some fancy `calc()` to solve this.

image

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserMenu.css.ts:23]: This causes a regression in application system form, where the user menu button is part of the grid but the modal appears in the corner.

Before:
image

After:
image

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (eirikurn) [inline libs/shared/components/src/auth/UserMenu/UserMenu.css.ts:26]: The modal now overflows in smaller windows, like small laptops and mobiles with some OS/browser chrome taking space from the website. The issue probably that there's now a fixed max-height for the delegation scroll-list while previously it flexed up and down to fill the screen vertically.

Before:

image

After:

image

**Review by eirikurn — COMMENTED** [review]
More UI comments.

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: Yeah I agree, let's find a solution to this 👍

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:121]: Hmm - I'll have to take a look at this - Is it because it's linked to directly? The breadcrumbs should stay.

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:146]: 100% agree - should be sm here

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.css.ts:26]: This is a problem! I'll talk to Keli and find a solution on smaller screens

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.css.ts:23]: Ohno - I'll fix

File: libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx:239]: Yes, that is the goal. Although I kinda agree when theres only one input. Maybe it can stay - that would solve this for now. Another thing to talk to Keli about.

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (eirikurn) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx:121]: Maybe it can’t show breadcrumbs unless it’s in the navigation tree? Should we have fake/hidden navigation entries?

File: libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx]: Done ✅

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx]: Done ✅

File: libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserProfileInfo.tsx]: Done ✅

File: libs/shared/components/src/auth/UserMenu/UserMenu.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.tsx]: Done ✅  - Fixed with lang button on desktop and lang select box on mobile

File: libs/shared/components/src/auth/UserMenu/UserDelegations.tsx
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserDelegations.tsx:82]: Done ✅ Fixed avatar background color and removed current actor from list.

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: Layout issues - The grid was wrong, was wider then it should be on the right side, it looks better for now. We are waiting for xxl breakpoint to be merged and then we will refactor this. 
Collapsed state - should stay like this for now at least

File: libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx
Comment (disaerna) [reply libs/service-portal/settings/access-control/src/screens/Access/components/AccessItem/AccessItem.tsx]: Done ✅

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.css.ts]: Done ✅

File: libs/shared/components/src/auth/UserMenu/UserMenu.css.ts
Comment (disaerna) [reply libs/shared/components/src/auth/UserMenu/UserMenu.css.ts]: Done ✅

**Review by maranomynet — APPROVED** [review]

**Review by arnorhs — APPROVED** [review]

File: libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx
Comment (oddsson) [inline libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx:251]: We probably don't want to render this heading as a p tag, because thats bad for accessibility. I think h3 headings should be rendered as h3 tags and h4 headings should be rendered as h4 tags.

File: libs/island-ui/core/src/lib/Input/Input.mixins.ts
Comment (oddsson) [inline libs/island-ui/core/src/lib/Input/Input.mixins.ts:70]: This seems a bit fishy. Do we not want to use the 8px grid here, i.e.

``padding: `${theme.spacing[1]}px, ${theme.spacing[2]}px, ${theme.spacing[1]}px, ${theme.spacing[1]}px` ``?

File: libs/island-ui/core/src/lib/Tag/Tag.tsx
Comment (oddsson) [inline libs/island-ui/core/src/lib/Tag/Tag.tsx:17]: We used to have this mint variant on tags but we removed it because the color contrast was not accessible (see here https://webaim.org/resources/contrastchecker/?fcolor=00B39E&bcolor=FFFFFF (this is the darkest variation of mint and it still fails.))

**Review by oddsson — CHANGES_REQUESTED** [review]
There are a few accessibility issues to consider in some of the island-ui files.

File: libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx
Comment (disaerna) [reply libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx]: whops - fixed

File: libs/island-ui/core/src/lib/Input/Input.mixins.ts
Comment (disaerna) [reply libs/island-ui/core/src/lib/Input/Input.mixins.ts:70]: Designer is taking a look at this one

File: libs/island-ui/core/src/lib/Tag/Tag.tsx
Comment (disaerna) [reply libs/island-ui/core/src/lib/Tag/Tag.tsx]: Ah i see - fixed

File: apps/service-portal/src/components/Layout/Layout.css.ts
Comment (darrikonn) [inline apps/service-portal/src/components/Layout/Layout.css.ts:10]: Might be better to specify what you're actually transitioning ... and add `will-change` as well so that the browser can optimize before the element is actually changed.

File: apps/service-portal/src/components/Layout/helpers.ts
Comment (darrikonn) [inline apps/service-portal/src/components/Layout/helpers.ts:65]: Nit-picky.. but hard to read ... can be simplified:
```typescript
const getLayoutType = () => {
  if (sidebarCollapsed) {
    return hasWideLayout ? 'wideClosed' : 'defaultClosed'
  }
  return hasWideLayout ? 'wide' : 'default'
}

return gridlayout[getLayoutType()]
```

File: apps/service-portal/src/components/Sidebar/Sidebar.css.ts
Comment (darrikonn) [inline apps/service-portal/src/components/Sidebar/Sidebar.css.ts:32]: Why are we not following the spacings from theme for this?

This might look inconsistent.

File: libs/island-ui/core/src/lib/Input/Input.mixins.ts
Comment (darrikonn) [reply libs/island-ui/core/src/lib/Input/Input.mixins.ts:70]: Ditto on this ^

Would also like to see it in the Button component.

File: libs/island-ui/core/src/lib/Select/Select.css.ts
Comment (darrikonn) [inline libs/island-ui/core/src/lib/Select/Select.css.ts:198]: theme spacings?

File: libs/island-ui/core/src/lib/Tag/Tag.tsx
Comment (darrikonn) [reply libs/island-ui/core/src/lib/Tag/Tag.tsx]: This can work if the font color is black and the background is lighter mint ... let's see what the design team thinks of that combination.

File: libs/service-portal/core/src/components/EducationCard/EducationCard.tsx
Comment (darrikonn) [inline libs/service-portal/core/src/components/EducationCard/EducationCard.tsx:36]: nit-picky ... but translations?

File: libs/service-portal/core/src/components/IntroHeader/IntroHeader.tsx
Comment (darrikonn) [inline libs/service-portal/core/src/components/IntroHeader/IntroHeader.tsx:42]: ditto

**Review by darrikonn — CHANGES_REQUESTED** [review]

**Comment by darrikonn** [comment]
Can you provide updated screenshots?
Have similar comments on the designs, but don't want to re-comment on something that has already been fixed/updated.

Design comment:
IMO this does not look good ... would probably look better with a lighter blue background color, or use the purple one like in the delegation dropdown:
image

File: libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx
Comment (thordurhhh) [reply libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx]: Won't that be an accessibility issue as well (a minor one sure) when you have perhaps a h1 or no heading at all and then the next heading is a h3 in the ActionCard? Would it make sense to allow the developer to use a `as` prop?

Comment (disaerna) [reply libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx]: Hmm good point -  @oddsson - what do you think?

File: libs/island-ui/core/src/lib/Input/Input.mixins.ts
Comment (disaerna) [reply libs/island-ui/core/src/lib/Input/Input.mixins.ts]: fixed

File: libs/island-ui/core/src/lib/Tag/Tag.tsx
Comment (disaerna) [reply libs/island-ui/core/src/lib/Tag/Tag.tsx]: Removed for now

File: apps/service-portal/src/components/Layout/Layout.css.ts
Comment (disaerna) [reply apps/service-portal/src/components/Layout/Layout.css.ts]: done

File: apps/service-portal/src/components/Layout/helpers.ts
Comment (disaerna) [reply apps/service-portal/src/components/Layout/helpers.ts]: thanks! fixed

File: apps/service-portal/src/components/Sidebar/Sidebar.css.ts
Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/Sidebar.css.ts]: fixed

File: libs/island-ui/core/src/lib/Select/Select.css.ts
Comment (disaerna) [reply libs/island-ui/core/src/lib/Select/Select.css.ts]: fixed

File: libs/service-portal/core/src/components/EducationCard/EducationCard.tsx
Comment (disaerna) [reply libs/service-portal/core/src/components/EducationCard/EducationCard.tsx]: fixed + other files as well

File: libs/service-portal/core/src/components/IntroHeader/IntroHeader.tsx
Comment (disaerna) [reply libs/service-portal/core/src/components/IntroHeader/IntroHeader.tsx]: fixed

File: apps/service-portal/src/components/Layout/Layout.css.ts
Comment (darrikonn) [inline apps/service-portal/src/components/Layout/Layout.css.ts:11]: Shouldn't this be `margin, flexBasis` ... since you're transitioning the margin + flexBasis.

**Review by darrikonn — CHANGES_REQUESTED** [review]
Minor misusage of willChange prop ... otherwise approved.

File: apps/service-portal/src/components/Layout/Layout.css.ts
Comment (disaerna) [reply apps/service-portal/src/components/Layout/Layout.css.ts]: whops - I've never seen this before 😄 fixed

File: apps/service-portal/src/components/Layout/Layout.css.ts
Comment (darrikonn) [inline apps/service-portal/src/components/Layout/Layout.css.ts:11]: This goes for the `transitions` prop as well ... not sure if I'm correct or not.

But I assume that values should follow CSS naming conventions (kebab) ... so this needs to be `flex-basis`.
Did the other one work?

Comment (disaerna) [reply apps/service-portal/src/components/Layout/Layout.css.ts]: lord help me, haha fixed 😅

**Review by darrikonn — APPROVED** [review]

**Review by oddsson — APPROVED** [review]

File: apps/service-portal/src/components/Sidebar/Sidebar.tsx
Comment (eirikurn) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: Sorry. This still looks weird to me.

Screenshot 2022-01-05 at 21 40 03

There is similar inconsistency in larger screens.

IMHO, our content container should always have the same spacing on left and right (to sidebar and page edge). On smaller screen sizes, we should use all available space except for 24px or whatever whitespace aligns with the overall page margin.

Comment (eirikurn) [reply apps/service-portal/src/components/Sidebar/Sidebar.tsx:21]: It looks like the horizontal whitespace of the page content is now spread over 3 elements. It would be nice if they could be simplified somehow. Happy to jump on a call to discuss the issue or brainstorm css for this.

**Review by eirikurn — APPROVED** [review]
Nice improvements all around. Submitting a bunch more design comments. But I think all of these are fairly minor. Happy to get this merged to main and fix rest in followup PRs. :tada:

File: apps/service-portal/src/components/Sidebar/NavItem/NavItem.tsx
Comment (darrikonn) [reply apps/service-portal/src/components/Sidebar/NavItem/NavItem.tsx:48]: The changes of `font-weight` on `hover` might also perform UI jumps to the next line if the text is long enough ... didn't test this myself, but is this being handled?

File: libs/shared/components/src/auth/UserMenu/UserDelegations.tsx
Comment (darrikonn) [reply libs/shared/components/src/auth/UserMenu/UserDelegations.tsx:73]: 20-100 delegations calls for a fuzzy search 🤔  ... might be worth checking in next projects

**Review by DavidBachmann — APPROVED** [review]

**Pull request merged**
